plugin/gcp/instance.py

```diff
@@ -0,0 +1,239 @@
+                self.logger.error(str(e))
+                raise GCPError(str(e))
+
+        return self.compute.instances().insert(
+            project=self.project,
+            zone=self.zone,
+            body=body).execute()
+
+    @blocking(True)
+    def delete(self):
+        """
+        Delete GCP instance.
+        Zone operation.
+
+        :return: REST response with operation responsible for the instance
+        deletion process and its status
+        """
+        self.logger.info('Delete instance')
+        return self.compute.instances().delete(
+            project=self.project,
+            zone=self.zone,
+            instance=self.name).execute()
+
+    @blocking(True)
+    def set_tags(self, tags):
+        """
+        Set GCP instance tags.
+        Zone operation.
+
+        :return: REST response with operation responsible for the instance
+        tag setting process and its status
+        """
+        # each tag should be RFC1035 compliant
+        self.logger.info('Set tags')
+        self.tags.extend(tags)
+        self.tags = list(set(self.tags))
+        fingerprint = self.get()["tags"]["fingerprint"]
+        return self.compute.instances().setTags(
+            project=self.project,
+            zone=self.zone,
+            instance=self.name,
+            body={"items": self.tags, "fingerprint": fingerprint}).execute()
+
+    @blocking(True)
+    def remove_tags(self, tags):
+        """
+        Remove GCP instance tags.
+        Zone operation.
+
+        :return: REST response with operation responsible for the instance
+        tag removal process and its status
+        """
+        # each tag should be RFC1035 compliant
+        self.logger.info('Remove tags')
+        self.tags = [tag for tag in self.tags if tag not in tags]
+        fingerprint = self.get()["tags"]["fingerprint"]
+        return self.compute.instances().setTags(
+            project=self.project,
+            zone=self.zone,
+            instance=self.name,
+            body={"items": self.tags, "fingerprint": fingerprint}).execute()
+
+    def get(self):
+        """
+        Get GCP instance details.
+        """
+        self.logger.info("Get instance details")
+        return self.compute.instances().get(
+            instance=self.name,
+            project=self.project,
+            zone=self.zone).execute()
+
+    @blocking(True)
+    def add_access_config(self):
+        """
+        Set GCP instance external IP.
+        Zone operation.
+
+        :return: REST response with operation responsible for the instance
+        external IP setting process and its status
+        """
+        body = {"kind": "compute#accessConfig",
+                "name": self.ACCESS_CONFIG,
+                "type": self.ACCESS_CONFIG_TYPE}
+        return self.compute.instances().addAccessConfig(
+            project=self.project,
+            instance=self.name,
+            zone=self.zone,
+            networkInterface=self.NETWORK_INTERFACE,
```

Comment:
It's the only way possible. Moreover GCP supports instance connected to only one network at the moment.